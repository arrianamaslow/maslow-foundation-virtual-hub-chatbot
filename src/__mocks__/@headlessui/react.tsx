/**
 * @jest-environment jsdom
 */

import { DialogProps } from '@headlessui/react'
import React, { ReactNode } from 'react'

type DialogPanelTestProps = {
  children?: ReactNode
}

interface DialogTestProps extends DialogProps {
  children?: ReactNode
}

const DialogPanel: React.FC<DialogPanelTestProps> = ({ children }) => {
  return <div data-testid='mock-dialog-panel-component'>{children}</div>
}

const Dialog: React.FC<DialogTestProps> = ({ open, children }) => {
  return (
    <div data-testid='mock-dialog-component'>
      {open ? 'DIALOGUEOPENED' : 'DIALOGUECLOSED'}
      {children}
    </div>
  )
}

export { Dialog, DialogPanel }
