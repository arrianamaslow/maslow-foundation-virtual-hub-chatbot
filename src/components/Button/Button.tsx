import style from './style.module.css'
export type ButtonProps = {
  text: string
  isDisabled?: boolean
  onClick: () => void
  buttonType: 'primary' | 'secondary' | 'tertiary'
}
const Button: React.FC<ButtonProps> = ({ text, isDisabled, onClick, buttonType }) => {
  return (
    <button
      className={`inline-flex w-full justify-center px-[24px] py-[10px] flex-shrink-0 rounded-lg border-solid border-[1px] text-[16px] ${style[buttonType]}`}
      onClick={onClick}
      disabled={isDisabled}
    >
      {text}
    </button>
  )
}
export default Button
