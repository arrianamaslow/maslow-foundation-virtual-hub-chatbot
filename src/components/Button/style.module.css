.primary {
  @apply border-black bg-black text-white hover:border-gray-500 hover:bg-gray-500 active:bg-stone-500 active:border-black;
}

.primary:disabled {
  @apply border-stone-300 bg-stone-300 text-white;
}

.secondary {
  @apply border-black bg-white hover:border-stone-300 active:border-black;
}

.secondary:disabled {
  @apply border-stone-300 bg-white text-stone-300;
}

.tertiary {
  @apply border-white bg-white hover:underline hover:text-stone-300 active:text-black active:underline;
}

.tertiary:disabled {
  @apply border-white bg-white text-stone-300;
}
