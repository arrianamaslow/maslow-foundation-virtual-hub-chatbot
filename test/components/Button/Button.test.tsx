/**
 * @jest-environment jsdom
 */

import { fireEvent, render, screen } from '@testing-library/react'
import Button, { ButtonProps } from '@/components/Button/Button'
import '@testing-library/jest-dom'
const defaultProps: ButtonProps = {
  text: 'placeholder text',
  isDisabled: false,
  onClick: jest.fn(),
  buttonType: 'primary'
}

describe('<Button />', () => {
  it('renders correctly', () => {
    const { container } = render(<Button {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

  it('renders differently for enabled and disabled', () => {
    const enabledPrimary = render(
      <Button {...defaultProps} isDisabled={false} buttonType='primary' />
    )
    const disabledPrimary = render(
      <Button {...defaultProps} isDisabled={true} buttonType='primary' />
    )

    const enabledSecondary = render(
      <Button {...defaultProps} isDisabled={false} buttonType='secondary' />
    )
    const disabledSecondary = render(
      <Button {...defaultProps} isDisabled={true} buttonType='secondary' />
    )

    const enabledTertiary = render(
      <Button {...defaultProps} isDisabled={false} buttonType='tertiary' />
    )
    const disabledTertiary = render(
      <Button {...defaultProps} isDisabled={true} buttonType='tertiary' />
    )

    expect(enabledPrimary.asFragment()).not.toEqual(disabledPrimary.asFragment())
    expect(enabledSecondary.asFragment()).not.toEqual(disabledSecondary.asFragment())
    expect(enabledTertiary.asFragment()).not.toEqual(disabledTertiary.asFragment())
  })

  it('renders differently for different button styles', () => {
    const enabledPrimary = render(
      <Button {...defaultProps} isDisabled={false} buttonType='primary' />
    )
    const enabledSecondary = render(
      <Button {...defaultProps} isDisabled={false} buttonType='secondary' />
    )
    const enabledTertiary = render(
      <Button {...defaultProps} isDisabled={false} buttonType='tertiary' />
    )

    const disabledPrimary = render(
      <Button {...defaultProps} isDisabled={true} buttonType='primary' />
    )
    const disabledSecondary = render(
      <Button {...defaultProps} isDisabled={true} buttonType='secondary' />
    )
    const disabledTertiary = render(
      <Button {...defaultProps} isDisabled={true} buttonType='tertiary' />
    )

    expect(enabledPrimary.asFragment()).not.toEqual(enabledSecondary.asFragment())
    expect(enabledSecondary.asFragment()).not.toEqual(enabledTertiary.asFragment())
    expect(enabledTertiary.asFragment()).not.toEqual(enabledPrimary.asFragment())

    expect(disabledPrimary.asFragment()).not.toEqual(disabledSecondary.asFragment())
    expect(disabledSecondary.asFragment()).not.toEqual(disabledTertiary.asFragment())
    expect(disabledTertiary.asFragment()).not.toEqual(disabledPrimary.asFragment())
  })

  it('fires event on click', () => {
    const clickEvent = jest.fn()
    render(<Button {...defaultProps} onClick={clickEvent} />)

    const buttonElement = screen.getByRole('button')
    fireEvent.click(buttonElement)

    expect(clickEvent).toHaveBeenCalled()
  })

  it('does not fire event when disabled', () => {
    const clickEvent = jest.fn()
    render(<Button {...defaultProps} isDisabled={true} onClick={clickEvent} />)

    const buttonElement = screen.getByRole('button')
    fireEvent.click(buttonElement)

    expect(clickEvent).not.toHaveBeenCalled()
  })
})
