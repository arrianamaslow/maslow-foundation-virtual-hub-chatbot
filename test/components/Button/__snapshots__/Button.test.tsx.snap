// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<div>
  <button
    class="inline-flex w-full justify-center px-[24px] py-[10px] flex-shrink-0 rounded-lg border-solid border-[1px] text-[16px] primary"
  >
    placeholder text
  </button>
</div>
`;
